src/auth/auth.service.spec.ts

```diff
@@ -93,14 +93,16 @@ describe('AuthService', () => {
         request.password,
         bcryptConstants.test_salt,
       );
-      jest.spyOn(userService, 'findOneUser').mockImplementation(() => {
-        return Promise.resolve({
-          id: faker.number.int(),
-          email: faker.internet.email(),
-          password: hashed_password,
-          role: Role.USER,
+      jest
+        .spyOn(userRepository, 'getUserWithPassword')
+        .mockImplementation(() => {
+          return Promise.resolve({
+            id: faker.number.int(),
+            email: faker.internet.email(),
+            password: hashed_password,
+            role: Role.USER,
```

Comment:
I'd use Pascal Case for enums if possible, so `Role.User`.

Also, here you are using enum but above you are passing normal string.